src/intrinsic/annotations.rs

```diff
@@ -65,19 +68,41 @@ impl AnnotationError {
                         "There must not be any statements after this annotated statement (and the annotated statement must not be nested in a block).",
                     ))
             }
+            AnnotationError::CalculusEncodingMismatch(span, calculus_name, encoding_name ) => {
+                Diagnostic::new(ReportKind::Error, span)
+                    .with_message(format!(
+                        "The '{}' calculus does not support the '{}' encoding.",
+                        calculus_name.name, encoding_name.name
+                    ))
+                    .with_label(Label::new(span).with_message(
+                        "The calculus and encoding are incompatible.",
+                    ))
+            }
+            AnnotationError::UnknownAnnotation(span, anno_name ) => {
+                Diagnostic::new(ReportKind::Error, span)
+                    .with_message(format!(
+                        "The '{}' annotation is unknown.",
```

Comment:
Does this error occur from the user? Or is it always a compiler error? If the latter is true, then we should simply panic instead of emitting an error if the user can't fix it.